### PR TITLE
Dynamics difference method for estimating eddy term

### PR DESCRIPTION
**Pull request by nbren12**
Implement the dynamics difference method for estimating the eddy flux term. This method computes the apparent source like this:

    Q = (high_res dyn - coarse dyn) + high_res physics 
        = high res storage - coarse dyn tendency 

This is different from the `apparent_sources_plus_dynamics_difference` method, which is not what I had in mind when originally proposing it. This approach doesn't directly use the eddy flux, so should be less susceptible to numerical discretization errors. 

I deleted the `apparent_sources_plus_dynamics_difference` method. @brianhenn I can revert the last commit if you were independently interested in that method.

### CODE REVIEW

File: external/loaders/loaders/mappers/_fine_res.py [hunk 1]
```diff
@@ -19,6 +20,15 @@ class MLTendencies(Protocol):
     dQ2: xr.DataArray
 
 
+class NudgedRun(Protocol):
+    tendency_of_air_temperature_due_to_dynamics: xr.Dataset
```
Comment (nbren12) [inline external/loaders/loaders/mappers/_fine_res.py:24]: @oliverwm1 Can you confirm that these are the names in your prescribed run outputs? Thanks.

File: external/loaders/loaders/mappers/_fine_res.py [hunk 2]
```diff
@@ -19,6 +20,15 @@ class MLTendencies(Protocol):
...
+    tendency_of_specific_humidity_due_to_dynamics: xr.Dataset
```
Comment (oliverwm1) [inline external/loaders/loaders/mappers/_fine_res.py:25]:
```suggestion
    tendency_of_air_temperature_due_to_dynamics: xr.DataArray
    tendency_of_specific_humidity_due_to_dynamics: xr.DataArray
```

File: external/loaders/loaders/mappers/_fine_res.py (hunk 1 above)
Comment (oliverwm1) [reply external/loaders/loaders/mappers/_fine_res.py:24]: Yes these are the names.

**Review by oliverwm1 — APPROVED** [review]
LGTM, very interested to see the results.

> This approach doesn't use the eddy flux at all, so should be less susceptible to numerical discretization errors.

It does indirectly use it, in that the eddy flux is used to drive the nudged+prescribed run which the coarse dynamics tendencies are coming from. I wonder how different the coarse dynamics are in a 'normal' nudged run.

**Comment by nbren12** [comment]
> It does indirectly use it, in that the eddy flux is used to drive the nudged+prescribed run which the coarse dynamics tendencies are coming from.

Good point. Would be nice to have another way to bootstrap the prescribed run. Maybe running without any dynamics difference estimate for the first prescribed run?

**Comment by nbren12** [comment]
@oliverwm1 Would you have time to run this configuration or point me to the configs you have been using?

**Pull request merged**